packages/azure-kusto-ingest/src/fileDescriptor.ts

```diff
@@ -58,8 +61,12 @@ export class FileDescriptor extends AbstractDescriptor implements FileDescriptor
         return path;
     }
 
-    async prepare(): Promise<string> {
-        if (this.zipped) {
+    async prepare(ingestionProperties?: IngestionPropertiesInput): Promise<string> {
+        if (ingestionProperties == null) {
```

Comment:
You are doing this twice, since you don't use the ingestionProperties again, you might as well just pass it to the next method as is